<launch>
  <arg name="debug" default="false" />
 
  <!-- <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="gui" value="$(arg gui)" />
    <arg name="world_name" value="$(arg world_file)"/>
  </include> -->

  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch">
    <arg name="port" value="9090"/>
  </include>
  
  <!-- <remap from="/cmd_vel_mux/input/teleop" to="/cmd_vel_mux/input/teleop"/> -->

  <include file="$(find youbot_gazebo_robot)/launch/youbot.launch"/>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find gatlin)/rviz/youbot_moveit.rviz" required="true" />

  <include file="$(find youbot_moveit_config)/launch/move_group.launch">
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="true"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>
  
  <node pkg="gatlin" type="remapper.py" name="remapper" output="screen"/> 

  <!-- <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node> -->
  
  <!-- Fake laser -->
  <!-- <node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
        args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
    <param name="scan_height" value="10"/>
    <param name="output_frame_id" value="/camera_depth_frame"/>
    <param name="range_min" value="0.45"/>
    <remap from="image" to="/camera/depth/image_raw"/>
    <remap from="scan" to="/scan"/>
  </node> -->
</launch>